client/reader/discover/components/navigation/v2/index.tsx

```diff
@@ -63,8 +66,11 @@ const DiscoverNavigationV2 = ( { selectedTab }: Props ) => {
 		},
 	];
 
+	// Only show the add new tab if the user is logged in.
+	const filteredTabs = baseTabs.filter( ( tab ) => tab.slug !== 'add-new' || isLoggedIn );
```

Comment:
This seems reasonable for the current case. Alternatively we could add isDisabled props to definitions to have a more general/extendable filter here, but we don't foresee any future need for that and can easily change it when needed.